packages/odie-client/src/data/use-get-zendesk-conversation.ts

```diff
@@ -1,28 +1,47 @@
 import Smooch from 'smooch';
 
+const parseResponse = ( conversation: Conversation ) => {
+	const messages = conversation?.messages.map( ( message ) => {
+		return {
+			content: message.text,
+			role: message.role,
+			type: message.type === 'text' ? 'message' : message.type,
+		};
+	} );
+
+	return { ...conversation, messages };
+};
 /**
  * Get the conversation for the Zendesk conversation.
  */
-export const getZendeskConversation = ( chatId: number | string | null | undefined ) => {
+export const getZendeskConversation = ( {
+	chatId,
+	conversationId,
+}: {
+	chatId: number | string | null | undefined;
+	conversationId?: string | null | undefined;
+} ) => {
 	if ( ! chatId ) {
 		return null;
 	}
 
 	const conversation = Smooch.getConversations().find( ( conversation ) => {
+		if ( conversationId ) {
+			return conversation.id === conversationId;
+		}
+
 		return Number( conversation.metadata[ 'odieChatId' ] ) === Number( chatId );
 	} );
 
 	if ( ! conversation ) {
 		return null;
 	}
 
-	const messages = conversation?.messages.map( ( message ) => {
-		return {
-			content: message.text,
-			role: message.role,
-			type: message.type === 'text' ? 'message' : message.type,
-		};
-	} );
-
-	return { ...conversation, messages };
+	// We need to ensure that more than one message is loaded
+	return Smooch.getConversationById( conversation.id )
```

Comment:
Is this `Smooch.getConversationById( conversation.id )` different than the `conversation` we retrieved before?

Reply:
It depends.
On first load all conversations will have only the last message in them.
This is to ensure that we retrieve the messages, and don't just display the last one.